### PR TITLE
Skip installer when not running on Windows

### PR DESCRIPTION
**Pull request by timbussmann**
When running on linux, the installer seems to usually trip over `Environment.OSVersion.Version.Major <= 5` and not run the installer (which is correct). The behavior does seem to be "lucky coincidence"  and instead we should avoid running the installer logic when not running on Windows.

### CODE REVIEW

**Comment by timbussmann** [comment]
@bording any thoughts on this given that Microsoft recommends to not use `OSVersion` when possible?

File: src/NServiceBus.Gateway/Installer/GatewayHttpListenerInstaller.cs [hunk 1]
```diff
@@ -34,6 +34,12 @@ public Task Install(string identity)
                 return Task.FromResult(0);
             }
 
+            if (Environment.OSVersion.Platform != PlatformID.Win32NT)
```
Comment (danielmarbach) [inline src/NServiceBus.Gateway/Installer/GatewayHttpListenerInstaller.cs]:
```suggestion
            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

File: src/NServiceBus.Gateway/Installer/GatewayHttpListenerInstaller.cs [hunk 2]
```diff
@@ -34,6 +34,12 @@ public Task Install(string identity)
...
+            {
+                logger.InfoFormat("Installer does not support platform {0}. Ensure that the process has required permissions to use listen to configured urls.", Environment.OSVersion.Platform.ToString("G"));
```
Comment (danielmarbach) [inline src/NServiceBus.Gateway/Installer/GatewayHttpListenerInstaller.cs]:
```suggestion
                logger.InfoFormat("Installer does not support platform {0}. Ensure that the process has required permissions to use listen to configured urls.", RuntimeInformation.OSDescription);
```

Comment (ramonsmits) [reply src/NServiceBus.Gateway/Installer/GatewayHttpListenerInstaller.cs]: I also removed "USE" from the string

**Pull request merged**